pkgs/development/libraries/qt-6/patches/qtdeclarative-include-store-paths-in-qmlc-paths.patch

```diff
@@ -0,0 +1,28 @@
+diff --git a/src/qml/jsruntime/qv4executablecompilationunit.cpp b/src/qml/jsruntime/qv4executablecompilationunit.cpp
+index 2ab38b70d9..40d1cefa47 100644
+--- a/src/qml/jsruntime/qv4executablecompilationunit.cpp
++++ b/src/qml/jsruntime/qv4executablecompilationunit.cpp
+@@ -33,6 +33,8 @@
+ #include <QtCore/qcryptographichash.h>
+ #include <QtCore/QScopedValueRollback>
+ 
++#include <QtCore/qcoreapplication.h>
++
+ static_assert(QV4::CompiledData::QmlCompileHashSpace > QML_COMPILE_HASH_LENGTH);
+ 
+ #if defined(QML_COMPILE_HASH) && defined(QML_COMPILE_HASH_LENGTH) && QML_COMPILE_HASH_LENGTH > 0
+@@ -74,6 +76,14 @@ QString ExecutableCompilationUnit::localCacheFilePath(const QUrl &url)
+     const QString localSourcePath = QQmlFile::urlToLocalFileOrQrc(url);
+     const QString cacheFileSuffix = QFileInfo(localSourcePath + QLatin1Char('c')).completeSuffix();
+     QCryptographicHash fileNameHash(QCryptographicHash::Sha1);
++
++    if (url.scheme() == QStringLiteral("qrc")) {
++        const QString applicationPath = QCoreApplication::applicationFilePath();
++
++        if (applicationPath.startsWith(QStringLiteral("/nix/store/")))
```

Comment:
Should probably have a comment here explaining why keying on just the binary store path works.